RadixWallet/Core/FeaturePrelude/FactorSourceCard/FactorSourceCard.swift

```diff
@@ -0,0 +1,295 @@
+					Text(subtitle)
+						.textStyle(.body2Regular)
+						.foregroundStyle(.app.gray2)
+				}
+
+				if let lastUsedOn = dataSource.lastUsedOn {
+					Text(
+						markdown: L10n.FactorSources.Card.lastUsed(RadixDateFormatter.string(from: lastUsedOn, dateStyle: .long)),
+						emphasizedColor: .app.gray2,
+						emphasizedFont: .app.body2Link
+					)
+					.textStyle(.body2Regular)
+					.foregroundStyle(.app.gray2)
+				}
+			}
+			.flushedLeft
+
+			if case let .selection(type, isSelected) = mode {
+				switch type {
+				case .radioButton:
+					RadioButton(
+						appearance: .dark,
+						state: isSelected ? .selected : .unselected
+					)
+				case .checkmark:
+					CheckmarkView(appearance: .dark, isChecked: isSelected)
+				}
+			}
+		}
+	}
+
+	struct LinkedEntitesView: SwiftUI.View {
+		@SwiftUI.State var isExpanded: Bool = false
+		let accounts: [Account]
+		let personas: [Persona]
+
+		var body: some SwiftUI.View {
+			VStack(alignment: .leading, spacing: .medium2) {
+				Button {
+					withAnimation(.default) {
+						isExpanded.toggle()
+					}
+				} label: {
+					HStack(spacing: .zero) {
+						Text(linkedTitle)
+							.textStyle(.body2Regular)
+							.foregroundStyle(.app.gray2)
+
+						Spacer(minLength: 0)
+
+						Image(isExpanded ? .chevronUp : .chevronDown)
+							.renderingMode(.original)
+					}
+				}
+
+				if isExpanded {
+					VStack(spacing: .small2) {
+						ForEach(accounts) { account in
+							AccountCard(kind: .compact, account: account)
+						}
+
+						ForEach(personas) { persona in
+							Card {
+								PlainListRow(
+									context: .compactPersona,
+									title: persona.displayName.value,
+									accessory: nil
+								) {
+									Thumbnail(.persona, url: nil)
+								}
+							}
+						}
+					}
+				}
+			}
+			.padding(.horizontal, .medium3)
+			.padding(.vertical, .small1)
+			.background(.app.gray5)
+		}
+
+		private var linkedTitle: String {
+			typealias Card = L10n.FactorSources.Card
+			let accountsString = accounts.count == 1 ? Card.accountSingular : Card.accountPlural(accounts.count)
+			let personasString = personas.count == 1 ? Card.personaSingular : Card.personaPlural(personas.count)
+
+			if accounts.count > 0, personas.count > 0 {
+				return Card.linkedAccountsAndPersonas(accountsString, personasString)
+			} else if accounts.count > 0 {
+				return Card.linkedAccountsOrPersonas(accountsString)
+			} else if personas.count > 0 {
+				return Card.linkedAccountsOrPersonas(personasString)
+			}
+			return ""
+		}
+	}
+}
+
+extension FactorSourceCard {
+	init?(
+		kind: Kind,
+		mode: Mode,
+		messages: [FactorSourceCardDataSource.Message] = [],
+		isExpanded: Bool = false,
+		onRemoveTapped: (() -> Void)? = nil
+	) {
+		guard
+			let icon = kind.factorSourceKind.icon,
+			let title = kind.title
+		else { return nil }
+
+		switch kind {
+		case let .genericDescription(factorSourceKind):
+			guard let details = factorSourceKind.details else { return nil }
```

Comment:
a factorSourceKind does not have details as per requirements or that we don't have these yet?

Reply:
made details not optional